src/java/org/apache/cassandra/cql3/functions/FormatFcts.java

```diff
@@ -0,0 +1,432 @@
+     * format_time(val, 'm') = format_time(val, 'ms', 'm')
+     * format_time(val, 's', 'm')
+     * format_time(val, 's', 'h')
+     * format_time(val, 's', 'd')
+     * format_time(val, 's') = format_time(val, 'ms', 's')
+     * format_time(val, 'h') = format_time(val, 'ms', 'h')
+     * </pre>
+     * <p>
+     * It is possible to convert values of a bigger unit to values of a smaller unit, e.g. this is possible:
+     *
+     * <pre>
+     * format_time(val, 'm', 's')
+     * </pre>
+     * <p>
+     * Values can be max of Double.MAX_VALUE, If the conversion produces overflown value, Double.MAX_VALUE will be returned.
+     * <p>
+     * Supported units are: d, h, m, s, ms, us, µs, ns
+     * <p>
+     * Supported column types on which this function is possible to be applied:
+     * <pre>INT, TINYINT, SMALLINT, BIGINT, VARINT, ASCII, TEXT</pre>
+     * For ASCII and TEXT types, text of such column has to be a non-negative number.
+     * <p>
+     * The conversion of negative values is not supported.
+     */
+    public static class FormatTimeFct extends NativeScalarFunction
+    {
+        private static final String FUNCTION_NAME = "format_time";
+
+        private FormatTimeFct(AbstractType<?>... argsTypes)
+        {
+            super(FUNCTION_NAME, UTF8Type.instance, argsTypes);
+        }
+
+        @Override
+        public ByteBuffer execute(Arguments arguments) throws InvalidRequestException
+        {
+            if (arguments.get(0) == null)
+                return null;
+
+            long value = validateAndGetValue(arguments);
+
+            if (arguments.size() == 1)
+            {
+                Pair<Double, String> convertedValue = convertValue(value);
+                return UTF8Type.instance.fromString(format(convertedValue.left) + ' ' + convertedValue.right);
+            }
+
+            TimeUnit sourceUnit;
+            TimeUnit targetUnit;
+            String targetUnitAsString;
+
+            if (arguments.size() == 2)
+            {
+                sourceUnit = MILLISECONDS;
+                targetUnitAsString = arguments.get(1);
+            }
+            else
+            {
+                sourceUnit = validateUnit(arguments.get(1));
+                targetUnitAsString = arguments.get(2);
+            }
+
+            targetUnit = validateUnit(targetUnitAsString);
+
+            double convertedValue = convertValue(value, sourceUnit, targetUnit);
+            return UTF8Type.instance.fromString(format(convertedValue) + ' ' + targetUnitAsString);
+        }
+
+        private TimeUnit validateUnit(String unitAsString)
+        {
+            try
+            {
+                return DurationSpec.fromSymbol(unitAsString);
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidRequestException(ex.getMessage());
+            }
+        }
+
+        private Pair<Double, String> convertValue(long valueToConvert) {
+            String[] units = {"d", "h", "m", "s"};
```

Comment:
these two could be private static members of this class so we would not need to create these arrays all over again